### PR TITLE
Privacy group updates and endpoints OR-278 OR-280

### PR DESCRIPTION
**Pull request by MadelineMurray**
Privacy group updates and endpoints

### CODE REVIEW

File: docs/Reference/API-Methods.md [hunk 1]
```diff
@@ -17,30 +17,100 @@ Content-Type: application/json
...
-    	"from": "4xanJzyaDPcBVMUSwl/tLp+DbXzd3jF9MKk1yJuyewE=",
-    	"to": ["YE5cJRJYTRO4XFo7yuAi/0K9DwjySGjsHB2YrFPnJXo="]
+      "addresses" : [
+          "A1aVtMxLCUHmBVHXoZzzBgPbW/wj5axDpW9X8l91SGo=",
+          "Ko2bVqD+nNlNYL5EE7y3IdOnviftjiizpjRt+HTuFBs="
+      ]
     }'
+    ``` 
+
+## privacyGroupId
```
Comment (iikirilov) [inline docs/Reference/API-Methods.md]:
```suggestion
## createPrivacyGroup
```

File: docs/Reference/API-Methods.md [hunk 2]
```diff
@@ -17,30 +17,100 @@ Content-Type: application/json
...
+        "from": "negmDcN2P4ODpqn/6WkJ02zT/0w0bjhGpkZ8UP6vARk="
+      }'
+    ```
+   
+    ```json tab="Result"
+    "68/Cq0mVjB8FbXDLE1tbDRAvD/srluIok137uFOaClOIn4m3Nn0P4jDuRGhWK3AAVdvkqQ==" 
+    ```
 
-`key` : *string* - Key used to receive the payload
+## findPrivacyGroupId
```
Comment (iikirilov) [inline docs/Reference/API-Methods.md]:
```suggestion
## findPrivacyGroup
```

File: docs/Reference/API-Methods.md [hunk 3]
```diff
@@ -17,30 +17,100 @@ Content-Type: application/json
...
+Content-Type: application/json
+
+**Request Body**
+
+`addresses` : *array of strings* - Public keys of Orion nodes for which to return privacy groups
 
 !!! example
     ```bash tab="curl HTTP request"
-    curl -X POST http://127.0.0.1:8888/send \
+    curl -X POST http://127.0.0.1:8888/findPrivacyGroupId \
```
Comment (iikirilov) [inline docs/Reference/API-Methods.md]:
```suggestion
    curl -X POST http://127.0.0.1:8888/findPrivacyGroup \
```

File: docs/Reference/API-Methods.md [hunk 4]
```diff
@@ -5,9 +5,9 @@ The Client API is used by Ethereum clients (for example, Pantheon) to interact w
 The port used by the Client API is defined by the `clientport ` property in the [configuration file](../Configuring-Orion/Configuration-File.md). 
 The default port is `8888`.
 
-## send
+## deletePrivacyGroupId
```
Comment (iikirilov) [inline docs/Reference/API-Methods.md]:
```suggestion
## deletePrivacyGroup
```

File: docs/Reference/API-Methods.md [hunk 5]
```diff
@@ -17,30 +17,100 @@ Content-Type: application/json
...
 
-`to` : *array of strings* - List of public keys to receive this payload
+**Returns**
 
+`privacyGroupId` : *string* - ID of the deleted privacy group 
 
-**Returns**
+!!! example
+    ```bash tab="curl HTTP request"
+    curl -X POST http://127.0.0.1:8888/deletePrivacyGroupId \
```
Comment (iikirilov) [inline docs/Reference/API-Methods.md]:
```suggestion
    curl -X POST http://127.0.0.1:8888/deletePrivacyGroup \
```

File: docs/Reference/API-Methods.md [hunk 6]
```diff
@@ -17,30 +17,100 @@ Content-Type: application/json
...
+
+`description` : *string* - Description for the privacy group. Optional. 
+
+**Returns**
+
+`privacy group` : *object* - Privacy group object 
+
+!!! example
+    ```bash tab="curl HTTP request"
+    curl -X POST http://127.0.0.1:8888/privacyGroupId \
```
Comment (iikirilov) [inline docs/Reference/API-Methods.md]:
```suggestion
    curl -X POST http://127.0.0.1:8888/createPrivacyGroup \
```

File: docs/Using-Orion/Privacy-Groups.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+When [transactions are sent](../Reference/API-Methods.md#send) to: 
+
+* List of recipients, Orion creates a privacy group containing the recipients 
+* Privacy group, the transaction is distributed to all members of the privacy group. 
+
+Pantheon maintains a [private state per privacy group](https://docs.pantheon.pegasys.tech/en/stable/Privacy/Privacy-Overview/#privacy-groups). 
+
+Endpoints are provided to create and manage privacy groups: 
+
+* [privacyGroupId](../Reference/API-Methods.md#privacygroupid)
```
Comment (iikirilov) [inline docs/Using-Orion/Privacy-Groups.md]:
```suggestion
* [createPrivacyGroup](../Reference/API-Methods.md#createPrivacyGroup) 
```

File: docs/Using-Orion/Privacy-Groups.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+* List of recipients, Orion creates a privacy group containing the recipients 
+* Privacy group, the transaction is distributed to all members of the privacy group. 
+
+Pantheon maintains a [private state per privacy group](https://docs.pantheon.pegasys.tech/en/stable/Privacy/Privacy-Overview/#privacy-groups). 
+
+Endpoints are provided to create and manage privacy groups: 
+
+* [privacyGroupId](../Reference/API-Methods.md#privacygroupid) 
+* [findPrivacyGroupId](../Reference/API-Methods.md#findprivacygroupid)
```
Comment (iikirilov) [inline docs/Using-Orion/Privacy-Groups.md]:
```suggestion
* [findPrivacyGroup](../Reference/API-Methods.md#findPrivacyGroup) 
```

File: docs/Using-Orion/Privacy-Groups.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+* List of recipients, Orion creates a privacy group containing the recipients 
+* Privacy group, the transaction is distributed to all members of the privacy group. 
+
+Pantheon maintains a [private state per privacy group](https://docs.pantheon.pegasys.tech/en/stable/Privacy/Privacy-Overview/#privacy-groups). 
+
+Endpoints are provided to create and manage privacy groups: 
+
+* [privacyGroupId](../Reference/API-Methods.md#privacygroupid) 
+* [findPrivacyGroupId](../Reference/API-Methods.md#findprivacygroupid) 
+* [deletePrivacyGroupId](../Reference/API-Methods.md#deleteprivacygroupid)
```
Comment (iikirilov) [inline docs/Using-Orion/Privacy-Groups.md]:
```suggestion
* [deletePrivacyGroup](../Reference/API-Methods.md#deletePrivacyGroup) 
```

File: docs/Reference/API-Methods.md [hunk 10]
```diff
@@ -99,68 +169,107 @@ To return the Privacy Group ID with the payload, use the `receive` method with t
...
     ```json tab="Result"
-    +3gnwO0oHXe4kXsr3kegd9jTTqsq3Y6Hm3w26WHR/RM=
+    Hello, World!
     ```
 
-## receiveraw
+## send
 
-Receives a raw payload from Orion using the payload key. The payload key is returned by the [sendraw](#sendraw) method
+Sends a payload to Orion.
```
Comment (Puneetha17) [inline docs/Reference/API-Methods.md]: We can also `send` using `privacyGroupId`. A sample request is as follows 
```
curl -X POST \
  http://127.0.0.1:9002/send \
  -H 'Content-Type: application/json' \
  -d '{
    "payload": "SGVsbG8sIFdvcmxkIQ==",
    "from": "A1aVtMxLCUHmBVHXoZzzBgPbW/wj5axDpW9X8l91SGo=",
    "privacyGroupId": "DyAOiF/ynpc+JXa2YAGB0bCitSlOMNm+ShmB/7M6C4zgmFoVyvJ5qxREuOQw5IQhe3IzeA=="
}'
```
The output from the request is same as that of `send`.

File: docs/Using-Orion/Privacy-Groups.md [hunk 11]
```diff
@@ -0,0 +1,17 @@
+# Privacy Groups 
+
+When [transactions are sent](../Reference/API-Methods.md#send) to: 
+
+* List of recipients, Orion creates a privacy group containing the recipients 
+* Privacy group, the transaction is distributed to all members of the privacy group. 
+
+Pantheon maintains a [private state per privacy group](https://docs.pantheon.pegasys.tech/en/stable/Privacy/Privacy-Overview/#privacy-groups). 
+
+Endpoints are provided to create and manage privacy groups:
```
Comment (Puneetha17) [inline docs/Using-Orion/Privacy-Groups.md]: We can only create and delete privacy groups for now and not manage.

File: docs/Reference/API-Methods.md [hunk 12]
```diff
@@ -17,30 +17,100 @@ Content-Type: application/json
...
+**Headers:**
+Content-Type: application/json
+
+**Request Body**
+
+`addresses` : *array of strings* - Public keys of Orion nodes to include in the privacy group
+
+`from` : *string* - Public key of the Orion node creating the privacy group
+
+`name` : *string* - Name of the privacy group. Optional.
```
Comment (Puneetha17) [inline docs/Reference/API-Methods.md]: `name` and `description` fields are not optional.

File: docs/Reference/API-Methods.md (hunk 4 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md]: Fixed

File: docs/Reference/API-Methods.md (hunk 5 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md]: Fixed

File: docs/Reference/API-Methods.md (hunk 2 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md]: Fixed

File: docs/Reference/API-Methods.md (hunk 3 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md]: Fixed

File: docs/Reference/API-Methods.md (hunk 1 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md]: Fixed

File: docs/Reference/API-Methods.md (hunk 12 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md]: Fixed

File: docs/Reference/API-Methods.md (hunk 6 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md]: Fixed

File: docs/Reference/API-Methods.md (hunk 10 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md]: Fixed

File: docs/Using-Orion/Privacy-Groups.md (hunk 11 above)
Comment (MadelineMurray) [reply docs/Using-Orion/Privacy-Groups.md]: Fixed

File: docs/Using-Orion/Privacy-Groups.md (hunk 8 above)
Comment (MadelineMurray) [reply docs/Using-Orion/Privacy-Groups.md]: Fixed

File: docs/Using-Orion/Privacy-Groups.md (hunk 9 above)
Comment (MadelineMurray) [reply docs/Using-Orion/Privacy-Groups.md]: Fixed

File: docs/Reference/API-Methods.md [hunk 13]
```diff
@@ -17,32 +17,119 @@ Content-Type: application/json
...
-    	"to": ["YE5cJRJYTRO4XFo7yuAi/0K9DwjySGjsHB2YrFPnJXo="]
-    }'
+        "privacyGroupId": "68/Cq0mVjB8FbXDLE1tbDRAvD/srluIok137uFOaClOIn4m3Nn0P4jDuRGhWK3AAVdvkqQ==",
+        "from": "negmDcN2P4ODpqn/6WkJ02zT/0w0bjhGpkZ8UP6vARk="
+      }'
     ```
    
     ```json tab="Result"
-    {"key":"wS+RMprLKIuCaHzOBfPeHmkJWUdOJ7Ji/9U3qj2jbXQ="}
+    "68/Cq0mVjB8FbXDLE1tbDRAvD/srluIok137uFOaClOIn4m3Nn0P4jDuRGhWK3AAVdvkqQ=="
```
Comment (iikirilov) [inline docs/Reference/API-Methods.md]: Is this an example of a privacy group id? It is the wrong length. This is the correct length `wS+RMprLKIuCaHzOBfPeHmkJWUdOJ7Ji/9U3qj2jbXQ=`

Comment (MadelineMurray) [reply docs/Reference/API-Methods.md]: Fixed

**Pull request merged**